components/header/header.tsx

```diff
@@ -1,8 +1,30 @@
 import Link from 'next/link';
 import Toggle from './burgerMenu';
 import Script from 'next/script';
+import { useEffect, useState } from 'react';
+import { useSession } from 'next-auth/react';
 
 function Header(props: any) {
+    const { data: sessions, status } = useSession();
+    const [isAdmin, setIsAdmin] = useState(false);
+
+    useEffect(() => {
+        if (status === 'authenticated') {
+            checkAdmin();
+        }
```

Comment:
I believe you can check if the user is admin from the client side itself, using the session data received from `useSession()` function

Reply:
it will save one api call to the backend by implementing the admin validation on client side itself